### DISCUSSION TITLE
Don't see the option for AMD in the Configure settings

### DISCUSSION THREAD

**Discussion by dginovker** [Acly/krita-ai-diffusion#813] [Q&A]
Hello and thank you for the project!

I have a discrete AMD Radeon RX 7700S which comes with 8GB of VRAM. I'm running Arch Linux with Krita and `python-virtualenv` installed via `pacman` and have plugin version 1.18.0, but I only see `Use CUDA (NVIDIA GPU)` and `Run on CPU` when I select `Local Managed Server`:

![Screenshot of Configure UI that is missing AMD](https://github.com/Acly/krita-ai-diffusion/assets/32943174/ec451859-95a9-46af-b3a7-ff0dd58a88a4)

I also have an integrated AMD Radeon 780M, is it possible this is causing problems?

**Comment by Acly** [comment]
No there's no integrated installer for AMD/Linux. You'll have to install ComfyUI yourself and use the Custom Server option.

**Comment by SyndicatedPillbug** [comment]
It would be really nice if there were to prioritize adding AMD support somehow. Installing ComfyUI on Linux and getting it all running smoothly defeats the lovely simplicity of just running this plugin in Krita. And NVIDIA GPUs are insanely expensive.

**Comment by Acly** [comment]
It would be really nice, but #76 has been open since the beginning of time. You either do it yourself, or keep waiting for someone with AMD hardware to do it.

**Comment by ricperry** [comment]
@Acly Would it be possible to just add a a ZLUDA (windows only) option using https://github.com/patientx/ComfyUI-Zluda rather than the "official" ComfyUI? I don't think anything else would need to change, but I'm not sure how this plugin sets up the venv. I've been trying to just activate the venv to modify the install myself, but I can't figure out how to do that.

**Reply by Acly** [reply]
According to the repo it requires a ton of manual setup, so I guess not

**Reply by lucaspeed72** [reply]
A detailed tutorial would be nice tho... I am one of the many human beings with a RX 9070 XT 16 Gb but installing and configuring ComfyUI by myself is way above my technical skills. I am running it on CPU for now but even with my 9950X3D it's a long task.

**Comment by ricperry** [comment]
@lucaspeed72 — if you're on Linux, switching PyTorch to ROCm is straightforward. I can’t speak to Windows. On Linux:

1. In the **Connection** tab, copy the **Server Path**, then open a terminal in that directory.

2. Activate the virtual environment:

   ```
   . ./venv/bin/activate
   ```

   Your prompt should show `(venv)`.

3. Replace the existing PyTorch packages:

   ```
   pip uninstall torch torchvision torchaudio
   pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/rocm7.2
   ```

   This replaces CUDA builds of PyTorch with ROCm builds.
   Current install options are listed here: [https://pytorch.org/get-started/locally/](https://pytorch.org/get-started/locally/)

**Note:** `torchaudio` is not always available for ROCm. That is typically not an issue for AI Diffusion workflows.

4. Restart Krita. The server should start normally.

**Additional note:** PyTorch may still report “CUDA” in some contexts; that does not necessarily reflect the actual backend in use.

---

**Commentary:**
ComfyUI has effectively been CUDA-only throughout its existence, despite the fact that ROCm works with minimal manual intervention. Not exposing ROCm as a standard install option is a gap in the project’s setup process.

**Reply by lucaspeed72** [reply]
Tried your precious info, I successfully installed everything:

'Successfully installed torch-2.11.0+rocm7.2 torchaudio-2.11.0+rocm7.2 torchvision-0.26.0+rocm7.2 triton-rocm-3.6.0'

But when I start Krita I can only start the server on CPU, trying with CUDA still tells me "No CUDA GPUs are available [1]"

A strange thing occur when the desktop session start, a message saying that a directory present in my .profile is missing (. "$HOME/.local/share/krita/ai_diffusion/server/uv/env")

**Reply by ricperry** [reply]
Have you confirmed you have the system-level ROCm libraries installed? They are installed by default on Fedora Workstation 43. 

Also, it won't work on WSL if you are on RDNA2 or older. ROCm on WSL is only compatible with RDNA3+.

**Reply by lucaspeed72** [reply]
I am on Mint 22.3, I have installed ROCm using amdgpu-install_7.2.1.70201-1_all.deb so I should be ok on that.
My card is a RX9070XT 16Gb and it should be RDNA4 afaik...

**Reply by ricperry** [reply]
There should be an open source version of the ROCm libraries. I never had success using the .deb from AMD's website.
